### DISCUSSION TITLE
Cleanup the issues?

### DISCUSSION THREAD

**Discussion by kugelfuhr** [cc65/cc65#2504] [General]
While looking through the issues I noticed that the list is not really in a good state.

* There are issues that have been fixed while still being open (example: #2110).
* There are issues marked as "bug" that aren't actually bugs (example: #2038).
* For some issues marked as bugs, a working code snippet to reproduce the bug is missing (example: #2465).

One solution might be to check and cleanup the list in regular intervals. Another to automatically close entries after some time and require them to be reopened if the problem isn't fixed. Would any of these be an option for the future? Or are there better ways to handle the problem?

**Comment by mrdudz** [comment]
The first and the third are still open because a test is needed. The other i have no idea, i don't have the time to read all that now.

And i am indeed checking the list every now and then. Automatically closing issues is a really really bad idea IMHO - most issues would be closed by now and thus the problems forgotten and the chance they ever get fixed will be even lower than it is now.

And the problem here is that simply no one is actively developing anymore. I hardly have the time to look at a few tickets once per week for that matter.

**Comment by polluks** [comment]
Maybe you've found a new assistant? ;-)

**Comment by kugelfuhr** [comment]
If the problem is fixed and it's just a test that is missing, why not remove the "bug" label and leave the "test_required" label? And if a compile/code generation problem is reported without an example to reproduce it, why not tag it with "maybe_bug" instead of "bug"?

I do of course understand your point and it is not my intention to stress you. I've just been looking through the list and found that it's very difficult to sort out "real" bugs (which require fixing) from "maybe bugs", "already fixed bugs" and "not really bugs" all of which have the "bug" label.

There might be many reasons for "no one is actively developing anymore" but having an issue list that makes it difficult to identify issues that need work will certainly not help.

But again: I do understand your lack of time and your point about development. Just because I do currently have some time to spare doesn't mean everybody has 😀

**Reply by rofl0r** [reply]
for someone that wants to spend time and effort on improving cc65, maybe the best target would be the to-do list for the next release, as it's been ages since the last release...

**Reply by mrdudz** [reply]
like https://github.com/cc65/wiki/wiki/Before-the-next-release ?

I am still leaning towards completely ditching "releases" though - we do not maintain a stable version anyway, so using the latest git head is pretty much recommended anyway.

**Reply by rofl0r** [reply]
most linux distros will always stick to "stable" releases no matter what, so unless an official release is made, you will always have people driving by reporting bugs that have long been fixed. imo in the long turn this costs more effort than making the occassional release. another advantage of having a stable release with a known feature set is that 3rd parties - such as game developers can do a preprocessor test for a certain cc65 version in order to have their code compile with whatever cc65 version a user might have installed.

**Reply by kugelfuhr** [reply]
I can only second the arguments regarding releases. Not having stable releases eases the life of the developers but makes life more difficult for all others.

Remember that developers using cc65 have users themselves. So someone writes a program and there are people compiling and using this program. With every cc65 change that breaks things, the program developer will get bug reports and has to handle them. And, handling them is not as easy as to simply say "use version x.y.z".

People have often problems getting their C programs to compile and run correctly with cc65 and you are adding the burden to compile specific versions of cc65 itself. It should be obvious that this definitely not user friendly.

My suggestion would be not to ditch stable releases but just ditch the idea that stable releases are something special and must contain a list of completed features. If there haven't been larger changes for - say - two months and there are no complaints whatsoever just stick a release tag to the current HEAD and be done.

**Comment by mrdudz** [comment]
Hence "make a github action that produces a release" on the TODO list. It should be as easy as pushing a tag, that triggers the action, and then produces tarballs and windows packages (perhaps even .debs).

> you will always have people driving by reporting bugs that have long been fixed

That hasn't been a problem in the past decade however. Most of the time you can be lucky that anyone would report anything at all :)

**Reply by clydeshaffer** [reply]
Well they report it to people who publish projects using cc65 instead :P

I get questions about the same issue every time someone tries my SDK because they assume their package manager will get them the right version.
